packages/strapi-parts/src/Combobox/Combobox.js

```diff
@@ -0,0 +1,274 @@
+    onOptionChange(index);
+    onOptionSelect(index);
+  };
+
+  const onOptionMouseDown = () => {
+    ignoreBlur.current = true;
+  };
+
+  const onOptionSelect = (index) => {
+    const selected = filteredNodes[index];
+    if (selected) {
+      onChange(selected.props.value);
+      return updateMenuState(false);
+    }
+
+    if (creatable) {
+      onCreateOption(inputValue);
+      updateMenuState(false);
+    }
+  };
+
+  const updateMenuState = (open, callFocus = true) => {
+    setOpen(open);
+    callFocus && inputRef.current.focus();
+  };
+
+  const filteredNodesClone = Children.toArray(filteredNodes).map((node, i) => {
+    const isActive = activeIndex === i;
+    return cloneElement(node, {
+      id: `${htmlId}-${i}`,
+      'aria-selected': selectedIndex === i ? 'true' : false,
+      ref: (r) => {
+        if (isActive) activeOptionRef.current = r;
+      },
+      onClick: () => onOptionClick(i),
+      onMouseDown: onOptionMouseDown,
+      isSelected: isActive,
+    });
+  });
+
+  return (
+    <>
+      <MainRow ref={containerRef} disabled={disabled}>
+        <Input
+          aria-activedescendant={activeId}
+          aria-autocomplete="list"
+          aria-controls={`${htmlId}-listbox`}
+          aria-expanded={`${open}`}
+          aria-haspopup="listbox"
+          aria-label={label}
+          aria-disabled={disabled}
+          readOnly={disabled}
+          ref={inputRef}
+          role="combobox"
+          type="text"
+          value={inputValue}
+          onBlur={!disabled ? onInputBlur : undefined}
+          onClick={!disabled ? () => updateMenuState(true) : undefined}
+          onInput={!disabled ? onInput : undefined}
+          onKeyDown={!disabled ? onInputKeyDown : undefined}
+          placeholder={placeholder}
+        />
+        <Row>
+          <CaretBox
+            disabled={disabled}
+            paddingLeft={3}
+            aria-hidden
+            as="button"
+            onClick={() => {
+              inputRef.current.focus();
+              updateMenuState(true);
+            }}
+            tabIndex={-1}
+          >
+            <DropdownIcon />
+          </CaretBox>
+        </Row>
+      </MainRow>
+      {open && (
+        <Popover
+          source={containerRef}
+          spacingTop={1}
+          fullWidth
+          intersectionId={`${htmlId}-listbox-popover-intersection`}
+          onReachEnd={hasMoreItems && !loading ? onLoadMore : undefined}
+        >
+          <div role="listbox" ref={listboxRef} id={`${htmlId}-listbox`}>
+            {Boolean(filteredNodes.length) ? (
+              filteredNodesClone
+            ) : creatable ? (
+              <OptionBox
+                paddingLeft={4}
+                paddingRight={4}
+                paddingTop={2}
+                paddingBottom={2}
+                ref={activeOptionRef}
+                onMouseDown={onOptionMouseDown}
+                role="option"
+                onClick={() => onOptionSelect()}
```

Comment:
```suggestion
                onClick={onOptionSelect}
```

Reply:
This fn accept a index param, We don't want to provide the event param here